openpaas-james/mailbox/encrypted/cassandra/src/main/java/com/linagora/openpaas/encrypted/cassandra/CassandraKeystoreDAO.scala

```diff
@@ -0,0 +1,74 @@
+package com.linagora.openpaas.encrypted.cassandra
+
+import java.nio.ByteBuffer
+
+import com.datastax.driver.core.querybuilder.QueryBuilder
+import com.datastax.driver.core.querybuilder.QueryBuilder.{asc, bindMarker, delete, insertInto, select}
+import com.datastax.driver.core.{PreparedStatement, Row, Session}
+import com.linagora.openpaas.encrypted.cassandra.table.KeyStoreTable.{ID, KEY, TABLE_NAME, USERNAME}
+import com.linagora.openpaas.encrypted.{KeyId, PublicKey}
+import javax.inject.Inject
+import org.apache.james.backends.cassandra.init.CassandraTypesProvider
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor
+import org.apache.james.core.Username
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+class CassandraKeystoreDAO @Inject()(executor: CassandraAsyncExecutor,
+                                     insertStatement: PreparedStatement,
+                                     selectStatement: PreparedStatement,
+                                     selectAllStatement: PreparedStatement,
+                                     deleteStatement: PreparedStatement,
+                                     deleteAllStatement: PreparedStatement) {
```

Comment:
Please don't pass statements in the constructor, you will not be able to initialize them with an inject.

Do directly vals in the class. CF CassandraTaskExecutionDetailsProjectionDAO